core/executor/add_liquidity_executor.go

```diff
@@ -58,7 +57,10 @@ func (e *AddLiquidityExecutor) Prepare() error {
 	// Mark the tree states that would be affected in this executor.
 	e.MarkLiquidityDirty(txInfo.PairIndex)
 	e.MarkAccountAssetsDirty(txInfo.FromAccountIndex, []int64{txInfo.GasFeeAssetId, txInfo.AssetAId, txInfo.AssetBId, txInfo.PairIndex})
-	liquidity := e.bc.StateDB().LiquidityMap[txInfo.PairIndex]
+	liquidity, err := e.bc.StateDB().GetLiquidity(txInfo.PairIndex)
+	if err != nil {
+		return err
```

Comment:
`Prepare` method also used by api. We can expose other errors, instead of internal `err`.

Reply:
I think we can return the internal error in the `APIProcessor`, please check https://github.com/bnb-chain/zkbnb/pull/187/commits/e14e49ab914f355164befee993b4d6d01939165e